pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -328,6 +328,45 @@ func (e *Controller) onServiceDelete(obj interface{}) {
 	e.queue.Add(key)
 }
 
+// onEndpointsAdd replaces checkLeftoverEndpoints()
+// on startup all current endpoints are added to the queue
+// and endpoints without a service associated will be removed
+// discard endpoints that are not managed by this controller
+// to avoid deleting custom endpoints
+func (e *Controller) onEndpointsAdd(obj interface{}) {
+	if !managedByController(obj.(*v1.Endpoints)) {
+		return
+	}
+
+	key, err := controller.KeyFunc(obj)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("Couldn't get key for object %+v: %v", obj, err))
+		return
+	}
+	e.queue.Add(key)
+}
+
+// onEndpointsUpdate handles the case when an user modifies an endpoint
+// so the controller can detect the change and set the correct status.
+func (e *Controller) onEndpointsUpdate(old, cur interface{}) {
+	key, err := controller.KeyFunc(cur)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("Couldn't get key for object %+v: %v", cur, err))
+		return
+	}
+
+	// Ignore endpoints that are going to be deleted
+	ep := cur.(*v1.Endpoints)
+	if !ep.DeletionTimestamp.IsZero() {
+		return
+	}
+
+	// Process the endpoint if it has been owned by this controller but modified by others
+	if managedByController(ep) && managedByChanged(ep) {
```

Comment:
Why not pass in old and current and then just return `managedBy(old) != managedBy(cur)` in the `managedByChanged` func?